lib/tasks/deploy.rake

```diff
@@ -134,12 +134,19 @@ namespace :deploy do
   end
 
   task :rolling_deploy do
+    stop_start_errors = []
     on_each_docker_host do |server|
       service = defined_service
 
-      stop_containers(server, service, fetch(:stop_timeout, 30))
-
-      container = start_new_container(server, service, defined_restart_policy)
+      begin
+        stop_containers(server, service, fetch(:stop_timeout, 30))
+        container = start_new_container(server, service, defined_restart_policy)
+      rescue e
+        on_fail = fetch(:rolling_deploy_on_failure, :exit)
+        raise unless on_fail == :continue
```

Comment:
Regarding the validation discussion, I'd greenlight this if a log line here made it obvious what happened. Something like 
```ruby
if on_fail == :continue
  info "Caught error #{e.message}, but continuing deploy because rolling_deploy_on_failure is #{on_fail}"
else
  error "Raising exception, as rolling_deploy_on_failure was #{on_fail} and not :continue"
  raise
end
```